skbio/stats/composition.py

```diff
@@ -1105,6 +1212,34 @@ def _gram_schmidt_basis(n):
     return basis.T
 
 
+def _build_basis(sbp):
+    """
+    Builds an orthogonal basis from a sequential binary partition. 
+       
+    "Code for a SBP. The first row represents the labels of parts of the compositional vector. At each order of
+    partition, +1 means that the part is assigned to the first group, −1 to the second group, and 0 that this part is not
+    in the group which is divided at this order."
+    J.J. Egozcue and V. Pawlowsky-Glahn, http://dugi-doc.udg.edu/bitstream/handle/10256/660/Egozcue-Pawlowsky.pdf
```

Comment:
I'm not entirely sure about what this method does, but it looks similar to the issue raised [here](https://github.com/biocore/scikit-bio/issues/1469).

If I have to guess, it is taking in a sequential binary partition, defining the contrasts at each internal node in the tree.  And you are trying to force a unit norm and orthogonality to each of the elements.  Is this correct?

If this method is to go into scikit-bio, then there should be some methods that make it easy to create sequential binary partitions for the ilr transform.  For example, I have a [method](https://github.com/biocore/gneiss/blob/master/gneiss/balances.py#L62) in gneiss that can convert a scikit-bio `TreeNode` object directly into an orthonormal basis, where the SBP is an intermediate calculation.  Its on my todo list to port this over to scikit-bio, but if you want to get started on it, you are more than welcome!